content/BCPatterns/event-bridge-pattern/index.md

```diff
@@ -0,0 +1,115 @@
++++
+title = "Event Bridge Pattern"
+weight = 1180
++++
+
+<_Created by waldo & Arend-Jan Kauffmann, Described by waldo_\>
+
+## Abstract
+
+In the world of interfaces, it is important to preserve (certain) events over multiple implementation of the interface.
+
+## Context
+
+An app can have interfaces.
+It makes it possible for other apps to extend/change the implementations of a certain part of the business logic.
+
+Like in this example, we have an interface, to implement different ways for getting weights from scales:
+
+```AL
+interface "IScale"
+{
+    procedure GetWeight(): Decimal;
+    procedure Tare();
+}
+```
+
+## Problem
+
+Multiple apps can subscribe to certain events of the app.
+When a new implementation is created, we need to make sure that these events are raised at the right times.  If those events were published on the implementation codeunit, it might very well be that those events will not be raised, hard to find, or whatever.
+
+So, if we would implement it like this, it isn't really extensible, as a different implemention would impement different events .. and it's not possible to subscribe to all of them (inculding future implementations)
```

Comment:
Typo: [...] as a different implemention would imp**l**ement different events [...] --- (impement to implement)
Typo: [...] to all of them (inc**lu**ding future implementations) --- (inculding to including)